Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+import os
+
+internal protocol EventStorage {
+    func startNewBatch()
+    func writeEvent(event: Event)
+    func batchReadyIds() -> [String]
+    func eventsFrom(id: String) -> [Event]
+    func remove(id: String)
+}
+
+internal class EventStorageImpl: EventStorage {
+    static let DIRECTORY = "events"
+    static let READYTOSENDEXTENSION = ".ready"
+    let encoder = JSONEncoder()
+    let decoder = JSONDecoder()
+    let folderURL: URL
+    var fileURL: URL
+    var currentBatch: [Event] = []
+
+    init() throws {
+        folderURL = URL(fileURLWithPath: try EventStorageImpl.getFolderURL())
+        fileURL = folderURL.appendingPathComponent("events-\(Date().currentTime)")
```

Comment:
if we already have a file we need to use that if not create a new one, in this way if we already have a file with events then if we create a new file we are going to miss this un-batched existing file.